x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/XPackSettings.java

```diff
@@ -245,11 +247,16 @@ public static Setting<String> defaultStoredHashAlgorithmSetting(String key, Func
     public static final String TRANSPORT_SSL_PREFIX = SecurityField.setting("transport.ssl.");
     private static final SSLConfigurationSettings TRANSPORT_SSL = SSLConfigurationSettings.withPrefix(TRANSPORT_SSL_PREFIX, true);
 
+    private static final SSLConfigurationSettings REMOTE_ACCESS_SSL = SSLConfigurationSettings.withPrefix(REMOTE_CLUSTER_SSL_PREFIX, false);
+
     /** Returns all settings created in {@link XPackSettings}. */
     public static List<Setting<?>> getAllSettings() {
         ArrayList<Setting<?>> settings = new ArrayList<>();
         settings.addAll(HTTP_SSL.getEnabledSettings());
         settings.addAll(TRANSPORT_SSL.getEnabledSettings());
+        if (TcpTransport.isUntrustedRemoteClusterEnabled()) {
+            settings.addAll(REMOTE_ACCESS_SSL.getEnabledSettings());
```

Comment:
are you sure we want to be able to disable TLS  for this port ?  ( for early days and ease of testing ...sure -> but long term ?) ..maybe a TODO to remove this ?
Also,  it is a bit confusing to why we treat this special here ... I would have expected the generalization of the `*.ssl ` to already handle this.

Reply:
I'm not sure what you mean? This is registering the "friendly" SSL settings if the feature flag is enabled - otherwise it's treated exactly like the HTTP and Transport SSL settings. I sort of assumed the idea was we would just look for all uses of `TcpTransport.isUntrustedRemoteClusterEnabled()` and remove them when we decide we don't need to lock it behind a feature flag anymore.

Reply:
This looks right to me. `getEnabledSettings()` actually has nothing to do with whether TLS can be disabled or not for this profile. That is controlled by a separate settings. This method just returns the list of available SSL settings if SSL is enabled. 
I have some doubts on how we actually enable the SSL for remote ingress. I'll comment on it separately.